lib/reek/spec/should_reek_of.rb

```diff
@@ -18,6 +18,7 @@ def initialize(smell_type_or_class,
         @smell_details = smell_details
         @configuration = configuration
         @configuration.load_values(smell_type => { Smells::SmellConfiguration::ENABLED_KEY => true })
+        @matching_smell_types = nil
```

Comment:
Yes, I think this case is not an instance of this smell, because no-one is **assuming** anything about the ivar.